crates/chain/tests/test_keychain_txout_index.rs

```diff
@@ -152,12 +156,29 @@ fn test_lookahead() {
             &[(external_descriptor.descriptor_id(), index)].into()
         );
 
-        assert_eq!(
-            txout_index.inner().all_spks().len(),
-            10 /* external lookahead */ +
-            10 /* internal lookahead */ +
-            index as usize + 1 /* `derived` count */
-        );
+        // test stored spks are expected
+        let exp_last_store_index = index + lookahead;
+        for i in index + 1..=exp_last_store_index {
+            assert_eq!(
+                txout_index.spk_at_index(TestKeychain::External, i),
+                Some(spk_at_index(&external_descriptor, i))
+            );
+        }
+        assert!(txout_index
+            .spk_at_index(TestKeychain::External, exp_last_store_index + 1)
+            .is_none());
+
+        // internal should only have lookahead
+        for i in 0..lookahead {
+            assert_eq!(
+                txout_index.spk_at_index(TestKeychain::External, i),
+                Some(spk_at_index(&external_descriptor, i))
```

Comment:
I still might need another pass to fully grasp the tests, but shouldn't this be asserting for the `TestKeychain::Internal`?

Reply:
Thank you, fixed